src/Support/FileRemover/FileBaseFileRemover.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Spatie\MediaLibrary\Support\FileRemover;
+
+use Illuminate\Contracts\Filesystem\Factory;
+use Spatie\MediaLibrary\MediaCollections\Filesystem;
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
+use Spatie\MediaLibrary\Support\FileRemover\FileRemover;
+use Spatie\MediaLibrary\Support\FileRemover\DefaultFileRemover;
+
+class FileBaseFileRemover extends DefaultFileRemover implements FileRemover
+{
+    public function __construct(protected Filesystem $mediaFileSystem, protected Factory $filesystem)
+    {}
+
+    public function removeAllFiles(Media $media): void
+    {
+        $this->removeFile($this->mediaFileSystem->getMediaDirectory($media). $media->file_name, $media->disk);
+
+        $this->removeConvertedImages($media);
+    }
+
+    public function removeConvertedImages(Media $media): void
+    {
+        $conversionsDirectory = $this->mediaFileSystem->getConversionDirectory($media);
+        $mediaDirectory = $this->mediaFileSystem->getMediaDirectory($media);
+
+        collect($media->getMediaConversionNames())->each(function ($conversionName) use ($media) {
```

Comment:
Shouldn't this use `getGeneratedConversions()` instead of `getMediaConversionNames()`?

Reply:
From memory, names are used to get to the specific file names individually